src/convertToObject.js

```diff
@@ -6,7 +6,35 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  // Transform string into array of lines
+  const propertiesArray = sourceString.split(';');
+  const styles = {};
+
+  // Transform separate lines of the array of properties into key-value pairs
+  propertiesArray.forEach((property) => {
+    const parts = property.split(':');
+
+    const key = dataCleanup(parts[0]);
+    const value = dataCleanup(parts[1]);
+
+    // Add to the resulting object only the lines with both key and value
+    if (key.length && value.length) {
+      styles[key] = value;
+    }
+  });
```

Comment:
[CODE PRACTICE] - Instead of using a `forEach` loop to create the `styles` object, consider using `reduce` to build the object in an immutable way. This aligns with the checklist item that suggests avoiding loops and using iteration methods instead.